apps/etterlatte-saksbehandling-ui/client/src/components/person/sakOgBehandling/SakStatus.tsx

```diff
@@ -4,44 +4,63 @@ import { Loader, Tag } from '@navikt/ds-react'
-    const sisteVedtak = hentStatusPaaSak(vedtakISak)
+  const visStatusPaaSisteVedtak = (vedtakISak: VedtakSammendrag[]): ReactNode => {
+    const sisteVedtak = hentSisteVedtak(vedtakISak)
 
-    switch (sisteVedtak?.vedtakType) {
-      case VedtakType.INNVILGELSE:
-        return (
-          <Tag key={VedtakType.INNVILGELSE} variant="success" icon={<RecordFillIcon aria-hidden color="#06893A" />}>
-            Løpende fra {!!sisteVedtak.virkningstidspunkt && formaterStringDato(sisteVedtak.virkningstidspunkt)}
-          </Tag>
-        )
-      case VedtakType.AVSLAG:
-        return (
-          <Tag key={VedtakType.AVSLAG} variant="error" icon={<XMarkIcon aria-hidden color="#C30000" />}>
-            Avslått den {!!sisteVedtak.virkningstidspunkt && formaterStringDato(sisteVedtak.virkningstidspunkt)}
-          </Tag>
-        )
-      case VedtakType.OPPHOER:
-        return (
-          <Tag key={VedtakType.OPPHOER} variant="alt2">
-            Ytelse opphørte den {!!sisteVedtak.virkningstidspunkt && formaterStringDato(sisteVedtak.virkningstidspunkt)}
-          </Tag>
-        )
-      default:
-        return (
-          <Tag key="annen-type" variant="warning">
-            Ubehandlet
-          </Tag>
-        )
+    if (!sisteVedtak) {
+      return (
+        <Tag key="annen-type" variant="warning">
+          Ubehandlet
+        </Tag>
+      )
     }
+
+    if (sisteVedtak?.vedtakType === VedtakType.AVSLAG) {
+      return (
+        <Tag key={VedtakType.AVSLAG} variant="error" icon={<XMarkIcon aria-hidden color="#C30000" />}>
+          Avslått den {sisteVedtak.virkningstidspunkt && formaterStringDato(sisteVedtak.virkningstidspunkt)}
+        </Tag>
+      )
+    }
+
+    const iDag = new Date()
+    const opphoerFraOgMed = sisteVedtak?.opphoerFraOgMed ? new Date(sisteVedtak.opphoerFraOgMed) : undefined
+
+    if (sisteVedtak?.vedtakType === VedtakType.OPPHOER || (opphoerFraOgMed && opphoerFraOgMed < iDag)) {
+      // TODO opphoerFraOgMed er nytt felt slik at opphørsvedtak som fantes før feltet vil ikke dette feltet populert
+      // Det skal populeres og når det er gjort kan vi anta her at et opphør alltid vil ha opphoerFraOgMed
+      const opphoerer = sisteVedtak.opphoerFraOgMed ? sisteVedtak.opphoerFraOgMed : sisteVedtak.virkningstidspunkt
+      return (
+        <Tag key={VedtakType.OPPHOER} variant="alt2">
+          Ytelse opphørte den {opphoerer && formaterStringDato(opphoerer)}
+        </Tag>
+      )
+    }
+
+    const foersteVedtak = hentFoersteVedtak(vedtakISak)
+
+    if (opphoerFraOgMed && opphoerFraOgMed >= iDag) {
+      return (
+        <Tag key={VedtakType.INNVILGELSE} variant="success" icon={<RecordFillIcon aria-hidden color="#06893A" />}>
+          Løpende fra {foersteVedtak?.virkningstidspunkt && formaterStringDato(foersteVedtak.virkningstidspunkt)} og
+          opphører {sisteVedtak.opphoerFraOgMed && formaterStringDato(sisteVedtak.opphoerFraOgMed)}
+        </Tag>
+      )
+    }
```

Comment:
Nå tror jeg det er alt for mye ren logikk sammen med rendringen av komponenter, samt at en del av sjekkene er vanskelig å lese. Flytt sjekkene i egne metoder i `sakStatusUtils.ts`